src/linker/Linker/LinkContext.cs

```diff
@@ -986,5 +988,17 @@ public enum CodeOptimizations
 		RemoveSubstitutions = 1 << 21,
 		RemoveLinkAttributes = 1 << 22,
 		RemoveDynamicDependencyAttribute = 1 << 23,
+
+		/// <summary>
+		/// Option to apply annotations to type heirarchy
+		/// Enable type heirarchy apply in library mode to annotate derived types eagerly
+		/// Otherwise, type annotation will only be applied with calls to object.GetType()
+		/// </summary>
+		OptimizeTypeHierarchyAnnotations = 1 << 24,
+
+		/// <summary>
+		/// Option to not special case EventSource
+		/// </summary>
+		RemoveEventSourceSpecialHandling = 1 << 25,
```

Comment:
For the special-case handling of serialization and custom operators, we made them separate command-line options instead of optimizations. Should we do the same here for consistency?

Reply:
Moved this away from as an optimization